hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/ReconServer.java

```diff
@@ -287,12 +287,17 @@ public void stop() throws Exception {
       if (ozoneManagerServiceProvider != null) {
         ozoneManagerServiceProvider.stop();
       }
-      if (reconDBProvider != null) {
-        reconDBProvider.close();
-      }
       if (reconTaskStatusMetrics != null) {
         reconTaskStatusMetrics.unregister();
       }
+      if (reconDBProvider != null) {
+        try {
+          LOG.info("Closing Recon Container Key DB.");
+          reconDBProvider.close();
+        } catch (Exception ex) {
+          LOG.error("Recon Container Key DB close failed", ex);
+        }
+      }
       isStarted = false;
```

Comment:
why not move it to the very end? mark service as stopped and then close the DB

Reply:
Ok thanks @kerneltime for reviewing the PR. We can do. However this marking service as stop is being used by above services which are already stopped before closing recon container DB. But no harm in moving to very end. Please re-review. Changes pushed as per suggestion.